File: Python/algoritms/codeSignals/rotate_matriz.py
import unittest


def rotate_matrix(a: list[list[int]]):

    a.reverse()
    for i in range(len(a)):
        for j in range(i):
            a[i][j], a[j][i] = a[j][i], a[i][j]
    return a


class test(unittest.TestCase):
    def matrix_test(self):

        a = [
            [1, 2, 3],
            [4, 5, 6],
            [7, 8, 9]
        ]

        expected = [[7, 4, 1],
                    [8, 5, 2],
                    [9, 6, 3]]
        self.assertEqual(expected, rotate_matrix(a))


if __name__ == "__main__":
    unittest.main()

File: rust/src/code_signals/rotate_matriz.rs
fn rotate_matrix(mut a: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    a.reverse();
    for i in 0..a.len() {
        for j in 0..i {
            (a[i][j], a[j][i]) = (a[j][i], a[i][j]);
        }
    }
    a
}
#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn test_matri() {
        let a: Vec<Vec<i32>> = vec![vec![1, 2, 3], vec![4, 5, 6], vec![7, 8, 9]];

        let expected: Vec<Vec<i32>> = vec![vec![7, 4, 1], vec![8, 5, 2], vec![9, 6, 3]];
        assert_eq!(expected, rotate_matrix(a))
    }
}
